airbyte-webapp/src/hooks/services/useConnectionHook.tsx

```diff
@@ -201,10 +202,49 @@ const useUpdateConnection = () => {
   return useMutation((connectionUpdate: WebBackendConnectionUpdate) => service.update(connectionUpdate), {
     onSuccess: (connection) => {
       queryClient.setQueryData(connectionsKeys.detail(connection.connectionId), connection);
+      // Update the connection inside the connections list response
+      queryClient.setQueryData<ListConnection>(connectionsKeys.lists(), (ls) => ({
+        ...ls,
+        connections:
+          ls?.connections.map((conn) => {
+            if (conn.connectionId === connection.connectionId) {
```

Comment:
This code might be slightly more readable if `connection` was called something like `updatedConnection` instead